src/transport_ofi.c

```diff
@@ -1297,6 +1445,12 @@ int shmem_transport_fini(void)
     if (shmem_transport_ofi_contexts) free(shmem_transport_ofi_contexts);
     shmem_transport_ctx_destroy(&shmem_transport_ctx_default);
 
+    for(i = 0; i < shmem_transport_ofi_stx_max; ++i) {
+        ret = fi_close(&shmem_transport_ofi_stx_pool[i].stx->fid);
+        OFI_CHECK_ERROR_MSG(ret, "STX context close failed (%s)\n", fi_strerror(errno));
+    }
+    free(shmem_transport_ofi_stx_pool);
```

Comment:
You should check that the hash table is empty and, if not, issue an error.  The user is required to free all private contexts before calling finalize.

Reply:
👍